sql/2020/01_CSS/custom_property_value_types.sql

```diff
@@ -0,0 +1,159 @@
+      "lightcyan", "lightgoldenrodyellow", "lightgray", "lightgreen", "lightgrey", "lightpink", "lightsalmon", "lightseagreen", "lightskyblue", "lightslategray", "lightslategrey",
+      "lightsteelblue", "lightyellow", "lime", "limegreen", "linen", "magenta", "maroon", "mediumaquamarine", "mediumblue", "mediumorchid", "mediumpurple", "mediumseagreen",
+      "mediumslateblue", "mediumspringgreen", "mediumturquoise", "mediumvioletred", "midnightblue", "mintcream", "mistyrose", "moccasin", "navajowhite", "navy", "oldlace",
+      "olive", "olivedrab", "orange", "orangered", "orchid", "palegoldenrod", "palegreen", "paleturquoise", "palevioletred", "papayawhip", "peachpuff", "peru", "pink", "plum",
+      "powderblue", "purple", "rebeccapurple", "red", "rosybrown", "royalblue", "saddlebrown", "salmon", "sandybrown", "seagreen", "seashell", "sienna", "silver", "skyblue",
+      "slateblue", "slategray", "slategrey", "snow", "springgreen", "steelblue", "tan", "teal", "thistle", "tomato", "turquoise", "violet", "wheat", "white", "whitesmoke",
+      "yellow", "yellowgreen",
+      "ActiveBorder", "ActiveCaption", "AppWorkspace", "Background", "ButtonFace", "ButtonHighlight", "ButtonShadow", "ButtonText", "CaptionText",
+      "GrayText", "Highlight", "HighlightText", "InactiveBorder", "InactiveCaption", "InactiveCaptionText", "InfoBackground", "InfoText",
+      "Menu", "MenuText", "Scrollbar", "ThreeDDarkShadow", "ThreeDFace", "ThreeDHighlight", "ThreeDLightShadow", "ThreeDShadow", "Window", "WindowFrame", "WindowText",
+      "transparent", "currentColor"
+    ];
+
+    const colorKeywordRegex = RegExp(`^(?:${colorKeywords.join("|")})$`, "gi");
+
+    walkElements(vars.computed, node => {
+      if (node.declarations) {
+        for (let property in node.declarations) {
+          let value;
+          let o = node.declarations[property];
+
+          if (property.startsWith("--")) {
+            // Custom property set to a value that doesn't depend on other variables
+            value = o.computed || o.value;
+            value = value.trim(); // whitespace is prserved in custom props
+          }
+          else {
+            incrementByKey(ret.properties, property);
+
+            if (o.references) {
+              let varCall = extractFunctionCalls(o.value, {names: "var"});
+
+              if (varCall.length === 1 && varCall[0].pos[1] === o.value.length) {
+                // The entire value is one var() call
+                value = o.computed || o.value;
+              }
+            }
+          }
+
+          if (value) {
+            incrementByKey(ret.values, value);
+
+            // What type of value is it?
+            let type;
+
+            let call = null;
+            // NOTE(rviscomi): Excluding XL values to avoid UDF timeouts.
+            if (value.length < 1000) {
+              call = extractFunctionCalls(value).filter(c => c.pos[0] === 0 && c.pos[1] === value.length)[0];
+            }
```

Comment:
@LeaVerou FYI this `extractFunctionCalls` was causing the UDF to timeout on the full dataset. I suspect it was a problem with some values being parsed incorrectly because capping the length to 1000 seems to have fixed it. LMK if this is ok or if there's a better way.

Reply:
Results in [this sheet](https://docs.google.com/spreadsheets/d/1sMWXWjMujqfAREYxNbG_t1fOJKYCA6ASLwtz4pBQVTw/edit?ts=5fad577d#gid=1306054386)

Reply:
Thanks! Yeah, that seems like a reasonable heuristic. Indeed, it seems common for Rework to parse multiple rules and declarations as one value, either due to syntax errors in the CSS, or due to parser bugs.

Reply:
As this is a custom metric, the parsing bug would have been in [css-variables.js](https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/css-variables.js), not Rework. Rework has definitely had similar bugs though.

Reply:
There is no logic to parse declarations there, so it must be a syntax error in the CSS.

Reply:
Gotcha. Thanks for clarifying.

Based on your review of the results, do you think there are any more SQL changes needed? If not I'll go ahead and merge this, otherwise I'll keep it open to catch any remaining changes. I think I've responded to all of your comments, though some still have open questions.